File: visual-model-ui/src/main/java/org/visual/model/ui/bar/MacOSTitleBar.java
package org.visual.model.ui.bar;


import javafx.event.ActionEvent;
import javafx.event.EventHandler;
import javafx.scene.control.Button;
import javafx.scene.input.MouseButton;
import javafx.scene.input.MouseEvent;

public class MacOSTitleBar extends CommonTitleBar {

    private Button closeButton = new Button("close");
    private Button sizeableButton = new Button("size");
    private Button minimizeButton = new Button("min");

    {
        closeButton.setOnAction(event -> close());

        sizeableButton.setOnMouseClicked(event -> {
            if (event.getButton() == MouseButton.PRIMARY && event.getClickCount() == 2) {
                restoreSizeOrMax();
            } else {
                maximizeWindow();
            }
        });

        minimizeButton.setOnAction(event -> minimizeWindow());
        getChildren().addAll(closeButton, sizeableButton, minimizeButton);
    }
}

File: visual-model-ui/src/main/kotlin/org/visual/model/ui/bar/MacosTitleBar.kt
package org.visual.model.ui.bar

import javafx.scene.control.Button

class MacosTitleBar : CommonTitleBar() {

    private val closeButton = Button("close").apply {
        setOnAction {
            close()
        }
    }
    private val sizeableButton = Button("size").apply {
        setOnMouseClicked {
            if (it.clickCount == 2) {
                restoreSizeOrMax()
            } else {
                maximizeWindow()
            }
        }
    }
    private val minimizeButton = Button("min").apply {
        setOnAction {
            minimizeWindow()
        }
    }

    init {
        children.addAll(closeButton, sizeableButton, minimizeButton)
    }
}
